src/main/java/seedu/address/logic/parser/SortTaskCommandParser.java

```diff
@@ -0,0 +1,36 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CRITERIA;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.SortTaskCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.task.DistinctTaskList;
+
+/**
+ * SortTaskCommandParser represents a parser which parses the arguments
+ * given by the user to create a SortTaskCommand object.
+ */
+public class SortTaskCommandParser implements Parser<Command> {
+    @Override
+    public Command parse(String args) throws ParseException {
+        ArgumentMultimap argumentMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_CRITERIA);
+        if (!isPrefixPresent(argumentMultimap)
+                || !argumentMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                    SortTaskCommand.MESSAGE_USAGE));
+        }
+        String criteria = argumentMultimap.getValue(PREFIX_CRITERIA)
+                .get();
+        if (!DistinctTaskList.isValidCriteria(criteria)) {
```

Comment:
Will it be better to create a Criteria class/enumeration to handle a criteria instead? Since it does not seem to be the responsibility of DistinctTaskList.

Reply:
I agree with you, I will look to change this!